File: questoesSlidesPLP/AddWhitespace.hs
addWhitespace:: String -> Int -> String
addWhitespace palavra 0 = palavra
addWhitespace palavra num = " " ++ addWhitespace palavra (num - 1)

main:: IO()
main =
  do
    palavra <- getLine :: IO String
    num <- readLn :: IO Int
    putStrLn(addWhitespace palavra num)

File: questoesSlidesProlog/addWhitespace.pl

addWhitespace(Palavra, Num, Resultado):- Num =:= 0,
  Resultado is Palavra.
addWhitespace(Palavra, Num, Palavra):- Num =/= 0,



main:-
  read(Palavra),
  read(Num),
  addWhitespace(Palavra, Num, Resultado),
  writeln(Resultado).